src/main/utils/channel/message-to-channel.injectable.ts

```diff
@@ -2,32 +2,24 @@
  * Copyright (c) OpenLens Authors. All rights reserved.
  * Licensed under MIT License. See LICENSE in root directory for more information.
  */
-import { lensWindowInjectionToken } from "../../start-main-application/lens-window/application-window/lens-window-injection-token";
-import { pipeline } from "@ogre-tools/fp";
 import { getInjectable } from "@ogre-tools/injectable";
-import { filter } from "lodash/fp";
 import { messageToChannelInjectionToken } from "../../../common/utils/channel/message-to-channel-injection-token";
 import type { MessageChannel } from "../../../common/utils/channel/message-channel-injection-token";
 import { tentativeStringifyJson } from "../../../common/utils/tentative-stringify-json";
+import getVisibleWindowsInjectable from "../../start-main-application/lens-window/get-visible-windows.injectable";
 
 const messageToChannelInjectable = getInjectable({
   id: "message-to-channel",
 
   instantiate: (di) => {
-    const getAllLensWindows = () => di.injectMany(lensWindowInjectionToken);
+    const getVisibleWindows = di.inject(getVisibleWindowsInjectable);
 
     // TODO: Figure out way to improve typing in internals
```

Comment:
This can be sort of fixed by 

1. Wrap the return value in `(...)` and then add an `as MessageToChannel` cast.

Also, I think it would be possible to simply the type `MessageToChannel` to the following:

```ts
export interface MessageToChannel {
  (channel: MessageChannel<void>): void;
  <Message>(channel: MessageChannel<Message>, message: Message): void;
}
```